lib/web_ui/test/text_editing_test.dart

```diff
@@ -598,6 +598,43 @@ void main() {
       expect(spy.messages, isEmpty);
     });
 
+    test('close connection on blur', () async {
+      final MethodCall setClient = MethodCall(
+          'TextInput.setClient', <dynamic>[123, flutterSinglelineConfig]);
+      textEditing.handleTextInput(codec.encodeMethodCall(setClient));
+
+      const MethodCall setEditingState =
+          MethodCall('TextInput.setEditingState', <String, dynamic>{
+        'text': 'abcd',
+        'selectionBase': 2,
+        'selectionExtent': 3,
+      });
+      textEditing.handleTextInput(codec.encodeMethodCall(setEditingState));
+
+      // Editing shouldn't have started yet.
+      expect(document.activeElement, document.body);
+
+      const MethodCall show = MethodCall('TextInput.show');
+      textEditing.handleTextInput(codec.encodeMethodCall(show));
+
+      checkInputEditingState(
+          textEditing.editingElement.domElement, 'abcd', 2, 3);
+
+      // DOM element is blurred.
+      textEditing.editingElement.domElement.blur();
+
+      expect(spy.messages, hasLength(1));
+      MethodCall call = spy.messages[0];
+      spy.messages.clear();
+      expect(call.method, 'TextInputClient.onConnectionClosed');
+      expect(
```

Comment:
It's also important to check that the `<input>` DOM element is properly cleaned up (it should be removed from the DOM).

Reply:
Interestingly: expect(document.getElementsByTagName('input'), hasLength(0)); fails in the beginning of the method.

I believe there is a problem with the teardown functions unrelated to this PR.

I'll address it in: https://github.com/flutter/flutter/issues/46891

Reply:
Hmmm there might be a leak somewhere. In any case, I was thinking of something more like:
```
expect(document.contains(textEditing.editingElement.domElement), isFalse);
```